torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -292,10 +292,12 @@ void ComputeAtMap::build(Fusion* fusion, GpuLower* gpu_lower) {
         for (auto entry : c2p_map) {
           auto c_id = entry.first;
           auto p_id = entry.second;
-          // If outside CA point and we're creating parallel map, do not map the
-          // axis
+          // If we're creating parallel map, do not map the axes when:
+          // 1) both of the axes are not leaf domains; or 2) the
+          // producer axis is outside CA point.
           if (mapping_mode_ == MappingMode::PARALLEL &&
-              right_of_ca_point.find(p_id) != right_of_ca_point.end()) {
+              (!(p_id->uses().empty() && c_id->uses().empty()) ||
```

Comment:
Makes me uneasy to check if there's no uses of the iteration domain. This should be safe, but seems like it could be error prone if for some reason we used an iteration domain in an operation but didn't update the domain. I think we should use the domain as the source of truth and have a separate for loop iterating across the consumer domain for this mapping.

Reply:
Thanks for the comment. I updated it accordingly.